bin/control.sh

```diff
@@ -31,134 +31,93 @@ fi
 #
 # Import common functions.
 #
-if [ "${IGNITE_HOME:-}" = "" ];
-    then IGNITE_HOME_TMP="$(dirname "$(cd "$(dirname "$0")"; "pwd")")";
-    else IGNITE_HOME_TMP=${IGNITE_HOME};
+if [ "${IGNITE_HOME:-}" = "" ]; then
+    IGNITE_HOME_TMP="$(dirname "$(cd "$(dirname "$0")"; pwd)")"
+else
+    IGNITE_HOME_TMP=${IGNITE_HOME}
 fi
 
-#
 # Set SCRIPTS_HOME - base path to scripts.
-#
 SCRIPTS_HOME="${IGNITE_HOME_TMP}/bin"
 
-source "${SCRIPTS_HOME}"/include/functions.sh
-source "${SCRIPTS_HOME}"/include/jvmdefaults.sh
+source "${SCRIPTS_HOME}/include/functions.sh"
+source "${SCRIPTS_HOME}/include/jvmdefaults.sh"
 
-#
-# Discover path to Java executable and check it's version.
-#
+# Discover path to Java executable and check its version.
 checkJava
 
-#
+# Determine Java version function
+determineJavaVersion() {
+    local version_output=$(java -version 2>&1)
+    local version=$(echo "$version_output" | awk -F '"' '/version/ {print $2}' | awk -F '.' '{if ($1 == 1) {print $2} else {print $1}}')
+    echo $version
+}
+
 # Discover IGNITE_HOME environment variable.
-#
 setIgniteHome
 
 if [ "${DEFAULT_CONFIG:-}" == "" ]; then
     DEFAULT_CONFIG=config/default-config.xml
 fi
 
-#
 # Set IGNITE_LIBS.
-#
-. "${SCRIPTS_HOME}"/include/setenv.sh
-. "${SCRIPTS_HOME}"/include/build-classpath.sh # Will be removed in the binary release.
```

Comment:
I don't think this line should be removed from the script. It is handled by release scripts when binary files/scripts are built. Could you please check it?

Reply:
Hi @sk0x50 ,

I've restored the line referencing `build-classpath.sh` as suggested. I initially removed it because this script is not present in the GridGain versions I've checked, and its inclusion triggers an error:

`./bin/control.sh: line 74: /opt/gridgain-community-8.8.39/bin/include/build-classpath.sh: No such file or directory
Control utility [ver. 8.8.39#20240403-sha1:a6d7ac2a]`

However, the main functionality of `control.sh` still works fine despite the error. Given your note that this script might be handled by release scripts when binaries are built, I assume you're aware of this issue.

Should we keep this line, accepting the error in some environments, or would you suggest a different approach?

Thanks for your guidance!

Reply:
Hi @dy2285 ,

Thanks for the clarification. Let me double check.

Reply:
Thanks for the information.